Configurations/50-masm.conf

```diff
@@ -11,9 +11,10 @@ my %targets = (
     "VC-WIN64A-masm" => {
         inherit_from    => [ "VC-WIN64-common", asm("x86_64_asm"),
                              sub { $disabled{shared} ? () : "x86_64_uplink" } ],
-        as              => "ml64",
-        asflags         => "/nologo /c /Cp /Cx /Zi",
-        asoutflag       => "/Fo",
+        AS              => "ml64",
+        ASFLAGS         => "/nologo",
+        ASOUTFLAG       => "/Fo",
+        asflags         => "/c /Cp /Cx /Zi",
```

Comment:
There is certain contradiction here. Idea is to let user override AS, but if user does that, then it's not given that non-overridable flags fit... One can even wonder if it would be appropriate to have possibility to specify non-overridable assembler. The situation is intensified by the fact that assembler can be associated with perlasm_scheme... Let's say this is a side note too...

Reply:
Yeah, I think I agree...  I did hesitate making this change, mind you, and would have no problem changing it back.

Reply:
Note that command attributes with names that match "make variables" except for casing are always seen as an alias and are overridable by default.  See [this](https://github.com/openssl/openssl/pull/5534/files#diff-f1206f9fadc5ce41694f69129aecac26R573) and [that](https://github.com/openssl/openssl/pull/5534/files#diff-f1206f9fadc5ce41694f69129aecac26R979) change in Configure.

Reply:
> Note that command attributes with names that match "make variables" except for casing are always seen as an alias and are overridable by default.

I do understand that. But this doesn't actually lifts contradiction between overridable assembler, non-overridable flags *and* perlasm_scheme. Well, there actually is similar contradiction even for compiler, because there also are non-overridable cflags and overridable CC. So that user is not free to override CC with compiler that doesn't understand non-overridable cflags. But there is no "intensifying factor" such as perlasm_scheme, and there is kind of "tradition" for multiple compilers mimicking most popular one. For example gcc and clang are interchangeable, cl and icc on Windows are interchangeable. But there are no interchangeable options for Windows assembler... Once again, it's side note :-)